FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -0,0 +1,428 @@
+
+                return JsonConvert.SerializeObject(_result);
+            }
+            catch (OperationCanceledException canceledEx)
+            {
+                // TODO: how about OperationCanceledException
+                _logger.LogInformation(canceledEx, "Job is canceled.");
+                throw;
+            }
+            catch (RetriableJobException retriableJobEx)
+            {
+                // always throw RetriableJobException
+                _logger.LogInformation(retriableJobEx, "Error in orchestrator job.");
+                throw;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation(ex, "Error in orchestrator job.");
+                throw new RetriableJobException("Error in orchestrator job.", ex);
+            }
+        }
+
+        // get the lastUpdated timestamp of next resource for next processing job
+        private async Task<DateTimeOffset?> GetNextTimestamp(DateTimeOffset? start, DateTimeOffset end, CancellationToken cancellationToken)
+        {
+            var typeFilters = _typeFilterParser.CreateTypeFilters(
+                _filterConfiguration.FilterScope,
+                _filterConfiguration.RequiredTypes,
+                _filterConfiguration.TypeFilters).ToList();
+
+            var parameters = new List<KeyValuePair<string, string>>
+            {
+                new (FhirApiConstants.LastUpdatedKey, $"lt{end.ToInstantString()}"),
+            };
+
+            if (start != null)
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"ge{((DateTimeOffset)start).ToInstantString()}"));
+            }
+
+            var searchOptions = new BaseSearchOptions(null, parameters);
+
+            var sharedQueryParameters = new List<KeyValuePair<string, string>>(searchOptions.QueryParameters);
+
+            DateTimeOffset? nexTimeOffset = null;
+            foreach (var typeFilter in typeFilters)
+            {
+                searchOptions.ResourceType = typeFilter.ResourceType;
+                searchOptions.QueryParameters = new List<KeyValuePair<string, string>>(sharedQueryParameters);
+                foreach (var parameter in typeFilter.Parameters)
+                {
+                    searchOptions.QueryParameters.Add(parameter);
+                }
+
+                var fhirBundleResult = await _dataClient.SearchAsync(searchOptions, cancellationToken);
+
+                // Parse bundle result.
+                JObject fhirBundleObject;
+                try
+                {
+                    fhirBundleObject = JObject.Parse(fhirBundleResult);
+                }
+                catch (Exception exception)
+                {
+                    _logger.LogError(exception, "Failed to parse fhir search result.");
+                    throw new FhirDataParseExeption("Failed to parse fhir search result", exception);
```

Comment:
For this case, we should add diagnostic log.

Reply:
Add TODO